src/openms/source/CHEMISTRY/ElementDB.cpp

```diff
@@ -101,6 +119,62 @@ namespace OpenMS
     buildElement_(name, symbol, an, abundance, mass);
   }
 
+  void ElementDB::addIsotope(const std::string& name,
+                     const std::string& symbol,
+                     const unsigned int an,
+                     double abundance,
+                     double mass,
+                     double half_life,
+                     Isotope::DecayMode decay,
+                     bool replace_existing)
+  {
+    unsigned int mass_number = std::round(mass);
+    string iso_name = "(" + std::to_string(mass_number) + ")" + name;
+    string iso_symbol = "(" + std::to_string(mass_number) + ")" + symbol;
+  
+    
+    if (!hasElement( an )  )
+    {
+      throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, String("Cannot add isotope ") +
+          name + " since the element with  atomic number " + an + " does not yet exsist -- create it first!");
+    }
+    
+  
+    if (hasElement( iso_symbol )  )
+    {
+      
+      if (!replace_existing)
+      {
+        throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, String("Isotope with symbol ") + iso_symbol + " already exists and replace_existing is set to false.");
+      }
+      
+      const Isotope* isotope = new Isotope(iso_name, iso_symbol, an, mass_number - an, mass, abundance, half_life, decay);      
+      addIsotopeToMaps_(iso_name, iso_symbol, isotope);
+      
+      // we changed/updated the isotopes, so we need to update
+      const Element* const_ele = isotope->getElement();
+      
+      Element* element = const_cast<Element*>(const_ele);
+      auto current_isotopes = element->getIsotopes();
+      
+      element->setIsotopes(element->getIsotopes());
```

Comment:
wow, this is hacky.
How about a member `refreshIsotopeCache()`, to at least show the intention.